boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts

```diff
@@ -76,6 +76,34 @@
 	scl-pin = <31>;
 };
 
+&spi0 {
```

Comment:
Could we do something like in line 47 for UARTE1?

Reply:
I.e. enable further SPIs based on used Kconfig configuration.

Reply:
Added. DTS intances of ID higher than 0 are now only added if enabled.

Reply:
The idea is that we should not need DTC to generate macros for things we do not aim to use. Not generating non-required defines always reduces risk of bugs. The same concept should be applied to .fixup files.

Now, I see that instances with ID 0 could also have conditional definitions (as long as the respective Kconfig options exist). We need to have a consistent approach, I agree.

Reply:
So what should I do under this pull request? I tried to be consistent with other peripherals configuration. @ioannisg please create a separate issue if you think all DTS instances should be added conditionally.

Reply:
I feel this conditional approach generates too much overhead in the DTS files. @ioannisg 
 eemember, that that the issue you described exists only for development kits. For application boards, there is usually a single useful hardware config: e.g. sensor1 connected to I2C0, sensor2 to SPI1.

On contrary, for development boards, you have a lot of unused peripherals and their usage change. However, in production the usage of every peripheral is clear. Application boards simply don't have DTS instances that are not needed. What do you think @pdunaj ?

Reply:
We should be avoiding #ifdef in .dts files as much as possible.  The general rule of thumb for the .dts is for it to enable by default what the board has enabled by default, as well as support for any headers.  @erwango has a writeup or such about what defaults we should be aiming for.

Reply:
Can you point to such writeup or a set of rules for writing DTS/Kconfig configurations?

Reply:
Hi, if I had to chose I would avoid adding ifdefs in dts file. If something is define but unused there is little harm in that (file is processed anyway, there will be few unused macros). On the contrary I think we should define everything that board has (is capable of), even if unused.